tests/testcontainers/logstash.py

```diff
@@ -23,128 +28,172 @@ class LogstashContainer(DockerContainer):  # type: ignore
+                auth_basic_password => "{self.logstash_password}"
+              }}
+            }}
+            output {{ stdout {{ codec => json_lines }} }}
+            """
+
+        self.with_env("LOG_LEVEL", os.environ.get("LOGSTASH_LOG_LEVEL", "fatal"))
         self.with_env("PIPELINE_WORKERS", "1")
         # By design and by default, Logstash does not guarantee event order. The following setting ensures that
         # events are processed in the order they are received.
         # Ref: https://www.elastic.co/guide/en/logstash/current/processing.html
         self.with_env("PIPELINE_ORDERED", "true")
-        self.with_env("CONFIG_STRING", "input { stdin {} } output { stdout {} }")
-        self.with_env("XPACK_MONITORING_ENABLED", "false")
+        self.with_env("MONITORING_ENABLED", "false")
+        self.with_env("CONFIG_STRING", logstash_config)
 
-    def get_url(self, ssl: bool = False) -> str:
+    def get_url(self) -> str:
         host = self.get_container_host_ip()
         port = self.get_exposed_port(self.port)
-        if ssl:
-            http_protocol = "https"
-        else:
-            http_protocol = "http"
-        return f"{http_protocol}://{host}:{port}"
+        return f"https://{host}:{port}"
 
-    def get_apiurl(self) -> str:
+    def get_api_url(self) -> str:
         host = self.get_container_host_ip()
         port = self.get_exposed_port(self.api_port)
         return f"http://{host}:{port}"
 
     @wait_container_is_ready(requests.exceptions.ConnectionError, requests.exceptions.ReadTimeout)  # type: ignore
     def _connect(self) -> None:
-        url = self.get_apiurl()
+        self.host = self.get_container_host_ip()
+        self.exposed_port = int(self.get_exposed_port(self.port))
+
+        while True:
+            try:
+                pem_server_certificate: str = ssl.get_server_certificate((self.host, self.exposed_port))
+                openssl_certificate = OpenSSLCrypto.load_certificate(
+                    OpenSSLCrypto.FILETYPE_PEM, pem_server_certificate.encode("utf-8")
+                )
+            except Exception:
+                time.sleep(1)
+            else:
+                self.ssl_assert_fingerprint = str(openssl_certificate.digest("sha256").decode())
+                break
+
+        assert len(self.ssl_assert_fingerprint) > 0
+
+        url = self.get_api_url()
         response = requests.get(f"{url}/?pretty", timeout=1)
         response.raise_for_status()
 
-    def start(self) -> "LogstashContainer":
+    def reset(self) -> None:
+        self._last_reset_message_count = self._previous_message_count
+
+    def start(self) -> LogstashContainer:
+        self._configure()
         super().start()
         self._connect()
         return self
 
-    def get_messages(self, retry: int = 2, timeout: int = 1, delay: int = 1) -> list[dict[str, Any]]:
+    def get_messages(self, expected: int, retry: int = 2, wait_time: int = 1, delay: int = 1) -> list[dict[str, Any]]:
```

Comment:
I've introduced the `expected` param in order to stabilise the reliability of the method

the previous logic was floating, considering we now run a single container where every integration test is run on, and I had multiple random failure related to the assertion on the results of this method